packages/server/src/api/controllers/user.js

```diff
@@ -4,27 +4,141 @@ const {
   getUserMetadataParams,
 } = require("../../db/utils")
 const { InternalTables } = require("../../db/utils")
-const { getGlobalUsers } = require("../../utilities/global")
+const { getGlobalUsers, getRawGlobalUser } = require("../../utilities/global")
 const { getFullUser } = require("../../utilities/users")
+const { isEqual } = require("lodash")
+const { BUILTIN_ROLE_IDS } = require("@budibase/auth/roles")
+const {
+  getDevelopmentAppID,
+  getAllApps,
+  isDevAppID,
+} = require("@budibase/auth/db")
+const { doesDatabaseExist } = require("../../utilities")
+const { UserStatus } = require("@budibase/auth/constants")
+
+async function rawMetadata(db) {
+  return (
+    await db.allDocs(
+      getUserMetadataParams(null, {
+        include_docs: true,
+      })
+    )
+  ).rows.map(row => row.doc)
+}
+
+function combineMetadataAndUser(user, metadata) {
+  // skip users with no access
+  if (user.roleId === BUILTIN_ROLE_IDS.PUBLIC) {
+    return null
+  }
+  delete user._rev
+  const metadataId = generateUserMetadataID(user._id)
+  const newDoc = {
+    ...user,
+    _id: metadataId,
+    tableId: InternalTables.USER_METADATA,
+  }
+  const found = Array.isArray(metadata)
+    ? metadata.find(doc => doc._id === metadataId)
+    : metadata
+  // copy rev over for the purposes of equality check
+  if (found) {
+    newDoc._rev = found._rev
+  }
+  if (found == null || !isEqual(newDoc, found)) {
+    return {
+      ...found,
+      ...newDoc,
+    }
+  }
+  return null
+}
+
+exports.syncGlobalUsers = async appId => {
+  // sync user metadata
+  const db = new CouchDB(appId)
+  const [users, metadata] = await Promise.all([
+    getGlobalUsers(appId),
+    rawMetadata(db),
+  ])
+  const toWrite = []
+  for (let user of users) {
+    const combined = await combineMetadataAndUser(user, metadata)
+    if (combined) {
+      toWrite.push(combined)
+    }
+  }
+  await db.bulkDocs(toWrite)
+}
 
-function removeGlobalProps(user) {
-  // make sure to always remove some of the global user props
+exports.syncUser = async function (ctx) {
+  let deleting = false,
+    user
+  const userId = ctx.params.id
+  try {
+    user = await getRawGlobalUser(userId)
+  } catch (err) {
+    user = {}
+    deleting = true
```

Comment:
This might give us some bother if there's ever a non 404 error here, accidentally marking the user as inactive. 

We could either 
- check for the error code to infer delete (easiest / fastest)
- send the action (create, update, delete) as a part of the request from worker -> server 
- send the user in the payload and don't read from the db (probably most future proof so that the database isn't coupled between services)

Probably option 1 is most suitable here, I'm mostly curious about: is removing the dependency on the same db entities across apis a longer term goal for us? Or are we fine with always giving multiple apis access to the same data. From experience it becomes an issue eventually and produced inconsistencies in the db.

Reply:
Good catch, i'll go with option 1 as I think that makes the most sense, just look for a 404, otherwise ignore. 

In terms of consistency, me and Marty discussed this, having to duplicate data across all of the databases isn't the nicest option but the only other option would be to remove access to the data within the apps, which I think would be a pretty breaking/brutal change. The issue comes down to searching/filtering, we need the data to be in the index, along with the actual user metadata (stored in the app) so that it can be used in a filter together.

I think it makes sense that some of this data should be duplicated, we also still do what we did before, going to the central DB to access the data when retrieving, so the data actually stored per app database is really just used for filtering, meaning there could be a filtering inconsistency, but when its actually retrieved it should be up to date.

Reply:
Chatted a bit about this offline. The main concerns is about reading and writing data from the `global` db in the app service, it seems we don't ever read from `app` databases in worker so it might be a good logical split to prevent cross domain direct db access like this in case service logic diverges, as it locks us in to using a shared lib which couples deployments of services.

Nothing required for this PR in particular, we'd like to start outlining the bigger ticket tech debt items to factor into future iterations. 

Some additional things we discussed:
- Write / read locking databases to particular services
- Moving apis into a single service (makes the above redundant)
- Moving automations into their own service